### PR TITLE
13-cartView - Added Cart View and CartController

### PR DESCRIPTION
**Pull request by matthamil**
## Status

**READY**
## Documentation

[X] I have fully documented features in this PR. Please put an `X` in the box to confirm.
## Migrations

NO
## Description

Users can now navigate to `localhost:5000/Cart/Index` to view his/her cart. Currently, the database is populated with dummy orders and products. When you navigate to the cart view, it should only show these products (dummy data has User with the ID of 1 logged in):

image

The `X` buttons to remove products from a cart **do not work**. The `Complete Order` button also **does not work**.
## Resolves Issue Number

Partially completes the following issues:

https://github.com/TeamCharles/initial-site/issues/13

https://github.com/TeamCharles/initial-site/issues/67
## Deploy Notes

Remove your `Migrations/` folder and delete `bangazon.db` and rebuild it. This must be done to add the new dummy data to your database from `DbInitializer.cs`.
## Steps to Test or Reproduce

``` sh
git pull --prune
git checkout 13-cartView
rm -rf Migrations
rm bangazon.db
dotnet ef migrations add Initial
dotnet ef database update
dotnet run
```

Navigate to `http://localhost:5000/Cart/Index`
## Impacted Areas in Application
- Views/Cart
- Controllers

### CODE REVIEW

File: Controllers/CartController.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+            float totalPrice = 0;
+            foreach (var product in activeProducts)
+            {
+                totalPrice += product.Price;
+            }
+
+            ViewBag.totalPrice = totalPrice;
+
+            if (activeProducts == null)
```
Comment (MattKraatz) [inline Controllers/CartController.cs]: Should probably move this above line 45?

File: Controllers/CartController.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            {
+                // TODO: This should probably be returning something other than `View()`.
+                return View();
```
Comment (MattKraatz) [inline Controllers/CartController.cs]: We can add a maybe redirect to a different view that just displays an empty cart message?

**Review by MattKraatz — APPROVED** [review]

**Comment by MattKraatz** [comment]
LGTM

**Comment by garrett-vangilder** [comment]
LGTM

**Comment by daynewright** [comment]
LGTM

**Pull request merged**